plugin/src/main/java/org/autorefactor/refactoring/ASTHelper.java

```diff
@@ -195,6 +196,32 @@ public int compare(ASTNode o1, ASTNode o2) {
         }
     }
 
+    private static final class VariableDeclarationIdentifierVisitor extends ASTVisitor {
+        private Set<String> variableNames = new HashSet<String>();
+        private ASTNode startNode;
+        private boolean includeInnerScopes;
+
+        private Set<String> getVariableNames() {
+            return variableNames;
+        }
+
+        private VariableDeclarationIdentifierVisitor(ASTNode node, boolean includeInnerScopes) {
+            startNode = node;
+            this.includeInnerScopes = includeInnerScopes;
+        }
+
+        @Override
+        public boolean visit(VariableDeclarationFragment node) {
+            variableNames.add(node.getName().getIdentifier());
+            return VISIT_SUBTREE;

```

Comment:
AFAIK, there are no new declarations below that, so it should `return DO_NOT_VISIT_SUBTREE`.